File: code.cpp
#include <iostream>
#include <string>
using namespace std;

string exampleFunction(bool arg1, string arg2)
{
  int num = 0x1F;
  char myChar = 'a';
  int year = 0;
  int age = 18;
  // This is a comment
  if (arg1)
  {
    year = 1453;
  }
  else
  {
    year = 2023;
  }

  switch (arg2[0])
  {
  case 'J':
    if (arg2 == "Jose" || arg2 == "Lydia")
    {
      age = 20;
    }
    else
    {
      age = 18;
    }
    break;
  case 'A':
    if (arg2 == "Angel" || arg2 == "Carolina")
    {
      age = 19;
    }
    else
    {
      age = 18;
    }
    break;
  default:
    age = 18;
    break;
  }

  string x = "hello";
  string y = "world";
  return x + ", " + y + "!";
}

void otherFunction()
{
  const int NUM = 1234;
  float decimal = 1.4;
  double d = .45;
  bool b1 = true;
  bool b2 = false;
  string str = "string";

  cout << NUM << endl;
  cout << decimal << endl;
  cout << d << endl;
  cout << b1 << endl;
  cout << b2 << endl;
  cout << str << endl;

  for (int i = 0; i < NUM; i++)
  {
    if (i == 10)
    {
      cout << "i equals 10!" << endl;
      break;
    }
    else if (i == 5)
    {
      continue;
    }
    cout << i << endl;
  }

  switch (NUM)
  {
  case 1:
    cout << "NUM is 1" << endl;
    break;
  case 2:
    cout << "NUM is 2" << endl;
    break;
  case 3:
    cout << "NUM is 3" << endl;
    break;
  default:
    cout << "NUM is not 1, 2, or 3" << endl;
    break;
  }

  int f = 1;
  int x = 10 - f;
  if (x > 5)
  {
    cout << "x is greater than 5" << endl;
  }
  else
  {
    cout << "x is not greater than 5" << endl;
  }

  try
  {
    int *ptr = new int[NUM];
    if (ptr == NULL)
    {
      throw "Memory allocation failed!";
    }
    delete[] ptr;
  }
  catch (const char *msg)
  {
    cout << msg << endl;
  }
}

int main()
{
  bool arg1 = true;
  string arg2 = "Jose";
  cout << exampleFunction(arg1, arg2) << endl;
  if (arg1 != false)
  {
    otherFunction();
  }
  int x = 10;
  int y = 3;
  int remainder = x % y; // remainder = 1
  return 0;
}

File: code.js
function exampleFunction(arg1, arg2) {
  let num = 0x1F;
  let year = 0;
  const age = 18; // default value
  // This is a comment
  if (arg1 == true) {
    year = 1453;
  } else {
    year = 2023;
  }
  switch (arg2) {
    case "Jose":
    case "Lydia":
      age = 20;
      break;
    case "Angel":
    case "Carolina":
      age = 19;
      break;
    default:
      break;
  }
  var x = 'hello';
  var y = 'world';
  return `${x}, ${y}! Age is ${age} and year is ${year}.`;
}
function otherFunction() {
  const NUM = 1234;
  const decimal = 1.4;
  const d = .45;
  const bool1 = true;
  const bool2 = false;
  const str = "string";

  for (let i = 0; i < NUM; i++) {
    if (i === 10) {
      console.log("i equals 10!");
      break;
    } else if (i === 5) {
      continue;
    }
    console.log(i);
  }

  switch (NUM) {
    case 1:
      console.log("NUM is 1");
      break;
    case 2:
      console.log("NUM is 2");
      break;
    case 3:
      console.log("NUM is 3");
      break;
    default:
      console.log("NUM is not 1, 2, or 3");
      break;
  }

  let f = 1;
  let x = 10 - f;
  if (x > 5) {
    console.log("x is greater than 5");
  } else {
    console.log("x is not greater than 5");
  }

  try {
    let arr = new Array(NUM);
    arr[0] = "Hello";
    arr[1] = "World";
    arr[2] = 1234;
    arr[3] = null;
    console.log(arr);
    throw "Something went wrong!";
  } catch (err) {
    console.log(err);
  }

  return;
}

function main() {
  var arg1 = true;
  var arg2 = "Jose";
  exampleFunction(arg1, arg2)
  if (arg1 != false) {
    otherFunction();
  }
  let x = 10;
  let y = 3;
  let remainder = x % y; // remainder = 1
}